net/httpclient.go

```diff
@@ -241,6 +267,11 @@ func NewTransport(options Options) *Transport {
 
 	htransport := &http.Transport{
```

Comment:
In retrospect it looks like it would be better to pass preconfigured `http.Transport` in the `Options` instead of duplication all its knobs...

Reply:
yes could be also a solution. Maybe I should create another NewClient and pass a http.Transport to support this and deprecate the other. wdyt?

Reply:
Or maybe lets have `*http.Transport` in the options such that when it is not nil it is used (otherwise new is created like before)?

Reply:
I.e. we can keep/deprecate existing options like `DisableKeepAlives` but if users need to configure e.g. DialContext they should provide a pre-configured transport

Reply:
makes sense!

Reply:
I changed the PR as you suggested